FreeRTOS/FreeRTOS
# Cleaned up report_builder in defender_demo_json

Description
-----------
With the parentheses in `DEFENDER_REPORT_SELECT_KEY` removed, json key macros from `defender.h` can now be used as strings to concatenate with other strings at build time. This PR cleans up `report_builder` in `defender_demo_json` by replacing "%s" in json formatting strings with appropriate key macros from `defender.h`.

Test Steps
-----------
This PR is tested by building defender_demo_json and running it. By default without `DEFENDER_USE_LONG_KEYS` defined, the json report is as follow:
```
{"hed": {"rid": 5658,"v": "1.0"},"met": {"tp": {"pts": [{"pt": 4456}],"t": 1},"up": {"pts": [],"t": 0},"ns": {"bi": 9635,"bo": 4085,"pi": 46,"po": 28},"tc": {"ec": {"cs": [{"lp": 4456,"rad": "54.213.72.205:8883"}],"t": 1}}},"cmet": {"stack_high_water_mark": [{"number": 57}],"task_numbers": [{"number_list": [2,5,4,1,3]}]}}
```
With `DEFENDER_USE_LONG_KEYS` defined to 1, the report is as follow:
```
{"header": {"report_id": 441,"version": "1.0"},"metrics": {"listening_tcp_ports": {"ports": [{"port": 33123}],"total": 1},"listening_udp_ports": {"ports": [],"total": 0},"network_stats": {"bytes_in": 7921,"bytes_out": 3670,"packets_in": 23,"packets_out": 25},"tcp_connections": {"established_connections": {"connections": [{"local_port": 33123,"remote_addr": "52.88.128.89:8883"}],"total": 1}}},"custom_metrics": {"stack_high_water_mark": [{"number": 57}],"task_numbers": [{"number_list": [2,5,4,1,3]}]}}
```



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
